lib/processExportsField.js

```diff
@@ -0,0 +1,270 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+/** @typedef {string|string[]} Mapping */
+/** @typedef {{[k: string]: ConditionalMapping|Mapping}} ConditionalMapping */
```

Comment:
```suggestion
/** @typedef {Record<string, ConditionalMapping | Mapping>} ConditionalMapping */
```

Reply:
typescript says Record can not circularly references itself